mbed_targets/_internal/targets_json_parsers/overriding_attribute_parser.py

```diff
@@ -93,11 +113,21 @@ def _determine_overridden_attributes(override_order: List[dict]) -> Dict[str, An
     Returns:
         A dictionary containing all the overridden attributes for a target
     """
-    target_attributes = reduce(lambda x, y: {**x, **y}, reversed(override_order))
+    target_attributes = _reduce_right_list_of_dictionaries(override_order)
+    for merging_attribute in MERGING_ATTRIBUTES:
+        override_order_for_single_attribute = map(lambda x: cast(dict, x.get(merging_attribute, {})), override_order)
```

Comment:
I find this name a bit confusing.
`override_order` indicates to me that it is a representation of ordering rules, how about `overrides_in_order` or `ordered_overrides`?

You could also use a list comprehension here - might be easier to read:
```
result: List[dict] = [x.get(merging_attributes, {}) for x in override_order]
```

Reply:
👍

Reply:
Gah naming is the most difficult. Not sure I have improved it much but fwiw - 905635a and 6d51d12.

Reply:
I think `targets_in_order` reads better than `accumulation_order`. 👍